stage1: fix saturating arithmetic on type comptime_int

riverbl opened the pull request:
Makes wrapping or saturating arithmetic operations map to their non wrapping or saturating equivalents when the operands have type comptime_int (previously only wrapping operations were mapped).

Makes saturating left shift map to exact left shift when the left operand has type comptime_int (previously saturating left shift was banned in this case).

Fixes #10365.

andrewrk reviewed (changes requested):
Test coverage please!

riverbl commented:
Fixed some issues with `BigInt` in stage1. Fixes the following bugs:

```zig
const std = @import("std");

pub fn main() void {
    std.debug.print("{}\n", .{ 593423721213448152027139550640105366508 - 7 == 593423721213448152027139550640105366501 });
}
```
Should print `true`, prior to fix printed `false` (printing large numbers of type `comptime_int` is still broken).

```zig
pub fn main() void {
    const foo: i8 = -128;
    const bar: i8 = -128;

    _ = foo +% bar;
}
```
Should compile, prior to fix compilation failed with "error: operation caused overflow".

Fixed stage2 bug where performing wrapping or saturating operations on type `comptime_int` resulted in the compiler reaching `unreachable`. Wrapping or saturating operations now map to their non wrapping or saturating equivalents.

Added some tests!

I think function `analyzeArithmetic` in Sema.zig should be refactored to move the code in each switch case to separate functions. If this is desired I'll add it to this pull request or submit another pull request.
